subprojects/distributions/src/integTest/groovy/org/gradle/TestDistributionIntegrationTest.groovy

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle
+
+import com.google.common.collect.Iterables
+import org.gradle.test.fixtures.file.TestFile
+
+import java.nio.file.Paths
+import java.util.zip.ZipEntry
+import java.util.zip.ZipFile
+
+class TestDistributionIntegrationTest extends  DistributionIntegrationSpec {
```

Comment:
The distributions integTests already have all the distributions as an [input](https://github.com/gradle/gradle/blob/9ee9bfd11c077e8da8915d7939614e6a9882077d/buildSrc/src/main/groovy/org/gradle/testing/DistributionTest.groovy#L98-L98). And I am sure that these are the only tests which rely on all of the distributions. You can check for usages of `IntegrationTestBuildContext#getDistributionsDir()`. There are some tests which rely on a distribution zip instead of the unpacked `intTestImage`. These are the tests which set `requiresBinZip = true` in their build files.